test/Test/Core/Issue.hs

```diff
@@ -20,10 +21,10 @@ import qualified Hedgehog.Range as Range
 import Test.Gen (genId, genRepoOwner, genRepoName)
 
 
-issueViaSql :: (WithDb env m, WithError m) => Issue -> m Issue
-issueViaSql issue = asSingleRow $ query
+issueViaSql :: (WithDb env m, WithError m) => WithId Issue -> m (WithId Issue)
+issueViaSql WithId{..} = asSingleRow $ query
     [sql| SELECT ?, ?, ?, ?, ?, ?, (? :: TEXT ARRAY) |]
-    (Only (issueId issue) :. issue)
+    ((Only withIdId) :. withIdVal)
```

Comment:
After you fix the `ToRow` instance for `WithId`, it should be possible to write this function as:

```haskel
issueViaSql = asSingleRow . query [sql| ... |]
```